cluster/gce/util.sh

```diff
@@ -589,6 +589,22 @@ function kube-up {
   find-release-tars
   upload-server-tars
 
+  if [[ ${KUBE_SHARE_MASTER:-} == "true" ]]; then
```

Comment:
Your refactoring and @fgrzadkowski's refactoring in #19218 look to have some overlap and some conflicts. We need to decide which one to merge first.